airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -3247,6 +3259,36 @@ def _format_schema_for_description(schema: dict) -> list:
     return description
 
 
+class ImpersonationToken:
+    """Simulate the interface of gcloud.aio.auth.token.BaseToken and generate impersonation_chain access_token."""
+
+    def __init__(self, project_id: str | None, impersonation_chain: str) -> None:
+        self.project_id = project_id
+        self.impersonation_chain = impersonation_chain
+
+    async def get_project(self) -> str | None:
+        project = (
+            self.project_id
+            or os.environ.get("GOOGLE_CLOUD_PROJECT")
+            or os.environ.get("GCLOUD_PROJECT")
+            or os.environ.get("APPLICATION_ID")
+        )
+        return project
+
+    async def get(self) -> str | None:
+        creds, _ = google.auth.default()
+
+        target_principal, _ = _get_target_principal_and_delegates(self.impersonation_chain)
+        impersonated_creds = impersonated_credentials.Credentials(
+            source_credentials=creds,
+            target_principal=target_principal,
+            target_scopes=["https://www.googleapis.com/auth/cloud-platform"],
+        )
+
+        impersonated_creds.refresh(google_auth_requests.Request())
```

Comment:
This is need's to be done asynchronously. Otherwise it will block the entire triggerer process.